Tests/Shared/Core/UnitTestBase.swift

```diff
@@ -14,7 +14,10 @@ import XCTest
 /// For SDK-wide tests, child classes can conveniently inherit from `UnitTest`, designed specifically for testing only SDK APIs.
 open class UnitTestBase<Component>: XCTestCase {
     public let testWriteKey = "test"
-    public var diGraphShared: DIGraphShared!
+    // Initialize DIGraphShared instantly to closely copy production behavior.
+    // The graph in production is initialized statically and can be accessed using .shared.
+    // This also allows us to override any dependency without waiting for SDK/modules to initialize and setup.
+    public var diGraphShared: DIGraphShared = .init()
```

Comment:
Updated mainly for `test_initialize_givenExistingIdentifier_expectGistSetProfileIdentifier`, but the comment above should help understand why I think it is good to initialize `DIGraphShared` early.

Reply:
If I understand the impact of this change, I disagree that we should make this change. 

I think this change would initialize the graph instance 1 time in the test class, instead of being re-initialized before every test function. 

If this change was to fix the `test_initialize_givenExistingIdentifier_expectGistSetProfileIdentifier` test, I would like to propose an alternative solution to make that test function correctly.

Reply:
Yes, it won't be re-initialized before every test function. But we do `reset` it on every `tearDown`. So far, this hasn't break anything. But if this can give us false results in future, we can surely try another approach to fix this specific test.

Reply:
Ah, I was only looking at `setup()` and did not see we are calling `reset` in `teardown`. 

Because we are calling `reset` between each test function running, I think my call-out wouldn't be an issue. 

If performing this `.init()` here causes side effects in our tests, I don't think that this is a good idea. With the context that you have with the test suite changes, if you can't think of any side effects of doing this change, then I am OK with resolving this.

Reply:
Sounds good. I also [slightly adjusted `tearDown`](https://github.com/customerio/customerio-ios/pull/506/commits/689e1e50d46cb7da6eeed0b03cb14e5757ac0e5c) to ensure graphs are reset to their exact initial state. This way, if any test fails, it's more likely due to an issue with the test itself or something that needs correction. Thanks for highlighting.